fix: enable DI for FeatureOptInService

eunjae-lee opened the pull request:
## What does this PR do?

This PR adds DI support for FeatureOptInService and FeaturesRepository.

But we're still using FeaturesRepository directly across the project, and I will fix it in a separate PR.

## Updates since last revision

- Renamed `FeatureOptInServiceInterface.ts` to `IFeatureOptInService.ts` to follow the codebase convention of using `I` prefix for interface files
- Renamed the interface from `FeatureOptInServiceInterface` to `IFeatureOptInService`
- Updated all references to use the new naming convention

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

1. Run type checks: `yarn type-check:ci --force`
2. Run tests: `TZ=UTC yarn test`
3. Verify the DI containers correctly resolve FeatureOptInService and FeaturesRepository

<!-- This is an auto-generated description by cubic. -->
---
## Summary by cubic
Enables DI for FeatureOptInService and FeaturesRepository across the app. Replaces direct instantiation and Prisma usage with moduleLoader-based containers and updates router/tests to resolve the service via DI.

- **Refactors**
  - Added DI modules/containers and helpers for FeatureOptInService and FeaturesRepository; introduced feature-specific DI tokens and merged them into DI_TOKENS.
  - Replaced the deprecated Features module with FeaturesRepository module and updated related imports.
  - Added IFeatureOptInService and implemented it in the service.
  - Updated TRPC viewer.featureOptIn router and integration tests to use DI.

<sup>Written for commit 259569fe4b454546661924de6ce04c380961e9d3. Summary will update on new commits.</sup>

<!-- End of auto-generated description by cubic. -->

---
Link to Devin run: https://app.devin.ai/sessions/de393c9ea2b8412587a8c4ca8cb2631d
Requested by: @eunjae-lee (eunjae@cal.com)

eunjae-lee committed:
fix integration test

eunjae-lee committed:
fix: enable DI for FeatureOptInService

eunjae-lee committed:
create containers/FeaturesRepository.ts

hariombalhara commented on packages/features/di/tokens.ts:
```diff
@@ -35,6 +35,8 @@ export const DI_TOKENS = {
   FEATURES_REPOSITORY: Symbol("FeaturesRepository"),
   FEATURES_REPOSITORY_MODULE: Symbol("FeaturesRepositoryModule"),
+  FEATURE_OPT_IN_SERVICE: Symbol("FeatureOptInService"),
+  FEATURE_OPT_IN_SERVICE_MODULE: Symbol("FeatureOptInServiceModule"),
```
The convention is to expose tokens.ts file from the feature package itself and just impor tthat here.  
  
See e.g.  https://github.com/calcom/cal.com/blob/rajiv/cal-7046-api-v2-teams-invite-link-endpoint/packages/features/oauth/di/tokens.ts#L8-L9

hariombalhara reviewed (changes requested):
Left a few suggestions

devin-ai-integration[bot] committed:
refactor: convert FeatureOptInService and FeaturesRepository to moduleLoader pattern

- Create feature-specific tokens in feature-opt-in/di/tokens.ts and flags/di/tokens.ts
- Update modules to use bindModuleToClassOnToken for type-safe dependency injection
- Simplify containers to use moduleLoader.loadModule() for automatic dependency loading
- Import feature-specific tokens in central tokens.ts using spread operator

Co-Authored-By: eunjae@cal.com <hey@eunjae.dev>

devin-ai-integration[bot] committed:
refactor: rename FeatureOptInServiceInterface to IFeatureOptInService

Follow the codebase convention of using 'I' prefix for interface files and names.

Co-Authored-By: eunjae@cal.com <hey@eunjae.dev>

eunjae-lee committed:
Merge branch 'main' into fix/di-feature-opt-in-service

devin-ai-integration[bot] committed:
fix: export featuresRepositoryModule for backward compatibility

The FeaturesRepository module was refactored to use moduleLoader pattern but
AvailableSlots.ts still imports featuresRepositoryModule. This adds the export
to maintain backward compatibility.

Co-Authored-By: eunjae@cal.com <hey@eunjae.dev>

hariombalhara reviewed (approved):
LGTM !!
